### PR TITLE
Weather API modification

### PR DESCRIPTION
**Pull request by technomooney**
I had to do something,,,, so I decided to start on the weather API module. 

i did a lot to the original file. @jk3132kv please take a look for sure... i assigned 2 others to review

### CODE REVIEW

**Review by jk3132kv — DISMISSED** [review]
Good work!

File: web_api/open_weather/test_open_weather_api.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+class TestPark():
+            def __init__(self):
+                self.lat = 44.59824417
+                self.lon = -110.5471695
+                self.park_code = "yell"
+
+class TestAPIProcessing(unittest.TestCase):
+
+    def test_extract_data_valid_json(self):
```
Comment (natalie-pratt) [inline web_api/open_weather/test_open_weather_api.py:16]: This test is failing, it seems somehow that the main or only difference between the two is the times. Is this a failure of the function or is it an issue with the example data given?

File: web_api/open_weather/open_weather_map.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+from datetime import datetime
+from pprint import pprint
+from collections import OrderedDict
+# Configure logging
+# logging.basicConfig(filename='weather.log', level=logging.INFO)
+
+# Read API key from environment variable
+api_key = os.environ['WEATHER_KEY']
+
+def get_api_response(lat,lon,url='https://api.openweathermap.org/data/2.5/forecast'):
```
Comment (natalie-pratt) [inline web_api/open_weather/open_weather_map.py:15]: Docstring here would be good!

File: web_api/open_weather/open_weather_map.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    try:
+        # Loop over forecast data and add data to the weather object
+        forecast = OrderedDict()
+        
+        for item in forecast_response:
+            date_time = datetime.fromtimestamp(item['dt']) # convert the unix timestamp to a datetime object
+            day_of_week = date_time.strftime("%A") # get the day that this specific entry is on
+            if day_of_week not in forecast.keys():
+                # Extract data from the forecast response and store it in the dictionary
+                # using the day of the week in as the dye and the time of day for each 3 hour section.
```
Comment (natalie-pratt) [inline web_api/open_weather/open_weather_map.py]: Is this comment supposed to say '...day of the week in as the dye...' here? I'm a little confused about what it means.

File: web_api/open_weather/test_open_weather_api.py (hunk 1 above)
Comment (technomooney) [reply web_api/open_weather/test_open_weather_api.py:16]: please pull and check again. i get all ok when i run the tests

File: web_api/open_weather/open_weather_map.py (hunk 3 above)
Comment (MCTCJulesO) [reply web_api/open_weather/open_weather_map.py]: Yea I am a little confused by that wording as well, maybe it is a misspelling?

**Review by MCTCJulesO — APPROVED** [review]
Looks good in terms of data I need for the database!

**Review by natalie-pratt — APPROVED** [review]
Looks good to merge!

**Pull request merged**